### PR TITLE
B 20740 and B 20296 ppm sit breakdown

### PR DESCRIPTION
**Pull request by antgmann**
## [B-20740](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20740) and [B-20296](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20296)

[INT PR](https://github.com/transcom/mymove/pull/13363)

## Summary

This feature adds a SIT Cost Breakdown similar to what is seen for service items for HHG SIT. This was originally planned to be a UI-only feature last PI, but had to be delayed and expanded due to PPM SIT items being classified and handled in the code as moving expenses rather than HHG's service items equivalent.

In order for this feature to work properly, an existing SIT Costing Estimate function on the closeout page was leveraged, with additional data from the pricing rate engine added to the API response.

With the appropriate data being sent back to the UI, a similarly-appearing replica of the HHG SIT calculations pane was recreated to display when the closeout counselor is reviewing calculated SIT Costs.

## Verification Steps for the Author

These are to be checked by the author.

- [x] Tested in the Integration environment
- [x] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### How to test

1. Access the client app as a new customer. 
2. Create and submit a PPM.
3. Log in as a service counselor. 
4. Approve the PPM.
5. Log back in as the service member. 
6. Upload your PPM Documents, including at least one Storage type item. 
7. Submit the PPM again for closeout.
8. Log back in as a service counselor. 
9. Approve the first weight ticket, until the Storage item is reached. 
10. Click the "Show calculations" button under the Government SIT Cost amount.
11. Verify all data is populating correctly while altering the storage length, amount, and location.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### CODE REVIEW

File: pkg/services/moving_expense/moving_expense_updater.go [hunk 1]
```diff
@@ -72,6 +72,7 @@ func (f *movingExpenseUpdater) UpdateMovingExpense(appCtx appcontext.AppContext,
 
 	if *mergedMovingExpense.MovingExpenseType == models.MovingExpenseReceiptTypeStorage &&
 		mergedMovingExpense.PPMShipment.Status == models.PPMShipmentStatusNeedsCloseout {
+
```
Comment (r-mettler) [inline pkg/services/moving_expense/moving_expense_updater.go:75]: 👀 neither is this empty line 😆

File: pkg/handlers/ghcapi/ppm_shipment.go [hunk 2]
```diff
@@ -148,6 +148,7 @@ func (h UpdatePPMSITHandler) Handle(params ppmsitops.UpdatePPMSITParams) middlew
 			}
 
 			ppmShipment.SITLocation = (*models.SITLocationType)(payload.SitLocation)
+
```
[the remark this answers is not in the document]
Comment (antgmann) [reply pkg/handlers/ghcapi/ppm_shipment.go:151]: Say the word and I'll take the shot

**Review by r-mettler — APPROVED** [review]
same as int

**Review by CoryKleinjanCACI — APPROVED** [review]
Matches INT.

**Pull request merged**